lib/graphlient/errors/connection_failed_error.rb

```diff
@@ -0,0 +1,10 @@
+module Graphlient
+  module Errors
+    class ConnectionFailedError < ServerError
+      def initialize(inner_exception)
+        @inner_exception = inner_exception
```

Comment:
@brianmcoates Do not assign the cause to `inner_exception` because [Ruby already does it automatically](https://www.honeybadger.io/blog/nested-errors-in-ruby-with-exception-cause/). Also, it is not always guaranteed that the response object exists when the cause of the failure is a connection failure. I would just translate the error to `Graphlient::Errors:: ConnectionFailedError` with doing nothing.

@ashkan18 The `inner_exception` pattern should be deprecated because people do the same thing over and over and it doesn't make sense to do things that Ruby does automatically.

Reply:
@yuki24 thank you for your feedback. I understand what you are saying I was just following the pattern that we had already in the gem @ashkan18 if you want me to remove it I can. :)

Reply:
Please do fix anything we wrote that's wrong, like in this case. We don't know what we're doing ;)

Reply:
haha awesome made the changes :)